### ISSUE TITLE
Why using GradientDescent as registration optimizer?

### ISSUE THREAD

**Issue by Anatole94** [confusius-tools/confusius#94]
Hi,
I saw that the optimizer used was GradientDescent.
You noted that the cross-correlation metrics works well with power Doppler data, is it the same for GradientDescent or just because it is the most basic optimizer?
Also default Matlab registration relies on RegularStepGradientDescent (which is robust to mean squares default metrics).
Thanks,

**Comment by sdiebolt** [comment]
Hi @Anatole94!

Gradient descent works well in practice for registration of most power Doppler data I've had my hands on. AFAIK, I haven't seen much of a difference with other optimizers, but that's something that would be interesting to benchmark.

In the future, I plan on adding more arguments to `register_volume` to allow tweaking more parameters from SimpleITK (MATLAB's registration functions, e.g. `imregtform`, is entirely based on ITK too, so parameters will be very similar). If you have a use for it, we could do a quick PR to allow using any of the [other optimizers that ITK offers](https://simpleitk.readthedocs.io/en/master/registrationOverview.html#optimizer)!

**Comment by Anatole94** [comment]
Thanks for the answer!
Actually I've added `RegularStepGradientDescent` to PyfUS to match `imregtform` for volumewise registration. And as you mentionned tuning the learning rate of 1e-4 gave a good results with default `GradientDescent` and` cross-correlation`.
Something else, I see that [you can mask the fixed volume](https://simpleitk.org/doxygen/latest/html/classitk_1_1simple_1_1ImageRegistrationMethod.html#a541df7e1ce5e69819ee190105e457d85), this would be useful to focus on ROI and speed up the process.

**Comment by sdiebolt** [comment]
I'll add a quick PR in the following days to add an `optimizer` argument. By the way, [you can already use fixed/moving masks in ConfUSIus](https://confusius.tools/latest/api/registration/#confusius.registration.register_volume(fixed_mask)) 😉

**Comment by Anatole94** [comment]
Wow that was spot on, I will copy this in my PyfUS version!
About adding optimizer, you will need to handle so many arguments, more versatile but less user-friendly.
But if you want to convert Matlab enthusiast, may be you can start with those used in MATLAB `imregtform`, we usually use the two preset of `imregconfig`.

**Comment by sdiebolt** [comment]
Indeed, the goal of ConfUSIus is to stay versatile! User-friendliness will be achieved through good defaults and heuristics so that these functions can be used with minimal arguments (and in the future, through an integration in the napari plugin/confusius CLI). But in the Python API, we should have all parameters available to allow for researchers to test things out and benchmarks the different methods!

Btw, remember to include our license in PyfUS if you copy from ConfUSIus (and also the license of the different tools we ourselves include, e.g. Nilearn, Brainglobe, SimpleITK, etc., depending on what you use!).

**Comment by Anatole94** [comment]
Ok ok.
I actually can't straight on copy but adapt it for PyfUS, is the ConfUSIus licensing still relevant?

**Comment by sdiebolt** [comment]
Yes it is! Any use, modification, adaptation, or distribution of openly licensed software must retain the license. If I remember correctly, PyfUS already includes the license from Nilearn and should include that from BrainGlobe since we implemented some of their logic. So that also applies to ConfUSIus if you adapt its code for use in PyfUS!

**Comment by Anatole94** [comment]
Ok understood, thanks!
